pkg/network/domainspec/generators.go

```diff
@@ -317,6 +321,37 @@ func (b *PasstLibvirtSpecGenerator) Generate() error {
 	iface := b.generateInterface(&b.domain.Spec.Devices.Interfaces[passtIfaceIdx])
 	b.domain.Spec.Devices.Interfaces[passtIfaceIdx] = *iface
 
+	go func() {
+		for {
+			if _, err := os.Stat(passtLogFile); !errors.Is(err, os.ErrNotExist) {
```

Comment:
Why the `os.Stat` is used? Isn't `os.Open` enough?

```
for {
		err := readFile(passtLogFile)
		if err != nil {
			if !os.IsNotExist(err) {
				log.Log.Reason(err).Errorf("failed to open file: %s", passtLogFile)
                                return
			} 
		}
		time.Sleep(interval)
	}
```

```
func readFile(fileName string) error {
	file, err := os.Open(fileName)
	if err != nil {
		return err
	}
	defer func() {
			if closeErr := file.Close(); closeErr != nil {
				log.Log.Reason(err).Errorf("failed to close file: %s", fileName)
			}
		}(file)
	// Read and process file content here
	// ...
	return nil
}

```

Reply:
The log file exist once the guest is booted.
Note that the for loop only checks if the log file exist, using `os.Stat` is more convenient for that purpose since there is no need to release resources. 
Once the file exist `os.Open` is used for reading, regarding [comment](https://github.com/kubevirt/kubevirt/pull/10326#discussion_r1306621577)  the file get opened once.

The log file is read once similar to the current implementation.
With the example you suggest passt logs will be append to virt-launcher logs repeatedly ending up with duplicated messages.

Reply:
Yes, you're right. I missed the fact is is open only once. How do you process messages that are added to the file after you finish the first read iteration?

> With the example you suggest passt logs will be append to virt-launcher logs repeatedly ending up with duplicated messages.
The file processing should take care of not printing duplicate content to the log.

Reply:
It would be easier to review this commit if the previous commit didn't remove the old log handling.

I thought this new commit adds a new log handling to solve the issue we discussed offline (the fact the file updates are not reflected in the log).
Please explain in the commit message/PR this issue you found and that it will be handled in a separate commit/PR.

Reply:
> Yes, you're right. I missed the fact it is opened only once. How do you process messages that are added to the file after you finish the first read iteration?
> 
> > With the example you suggest passt logs will be append to virt-launcher logs repeatedly ending up with duplicated messages.
> 
> The file processing should take care of not printing duplicate content to the log.

This will require working with file watchers to handling new log messages after the guest is booted.
The current implementation works like that (acts as `cat logsfile..`), this is true for other kind of logs: libevirtd, seabios, virtiofs.
This PR change just surface the problem, we can fix it in a follow up as you suggest.

> It would be easier to review this commit if the previous commit didn't remove the old log handling.
> 
> I thought this new commit adds a new log handling to solve the issue we discussed offline (the fact the file updates are not reflected in the log). Please explain in the commit message/PR this issue you found and that it will be handled in a separate commit/PR.

Sorry for the confussion I will squash that commit.
And update the PR desc with the logs issue.